Remove Errors section from var_os docs

fleetingbytes opened the pull request:
Remove `Errors` section from `var_os` documentation, fixes #109893

fleetingbytes committed:
Update env.rs

Remove `Errors` section from `var_os` documentation

Nilstrieb commented:
While it can't return an error, it can still return `None`, so I guess the docs should be changed to mention that instead. (yay, return type consistency between `var` and `var_os` :(. )

fleetingbytes commented:
@ Nilstrieb the documentation already mentions that `var_os` returns `None` if it is not present there is other errors. But additionally it currently says that `var_os` "returns an error if" which that can't be true, since its return type is Option.

ChrisDenton commented:
Right, the wording should be more like "on Error returns `None`" and then list the things that might cause it to return `None`.

I don't think removing the section entirely is a good idea. It contains useful information even if it mixes up the return type.

fleetingbytes committed:
add situation where var_os returns None

Re-introduced some of the former errors as situations where `None` is returned.

fleetingbytes commented:
> I don't think removing the section entirely is a good idea. It contains useful information even if it mixes up the return type.

Good point. I have reworded the documentation. The original Errors are now reformulated as situations where None is returned. (The error about invalid Unicode is gone for good, though.)

fmease commented on library/std/src/env.rs:
```diff
@@ -236,22 +236,18 @@ fn _var(key: &OsStr) -> Result<String, VarError> {
+///  - the environment variable's name contains
+/// the equal sign character (`=`) or the NUL character
+///
+/// If this is not desired, consider using [`var_os`].
```
```suggestion
```

Not sure what this line is supposed to convey. We already are at `var_os`, so why link to itself? Is it supposed to be `var`? If so, it has basically same behavior in regards to *not present*, `=` in value, `\0` in value.

fleetingbytes committed:
remove self-reference in var_os doc

Co-authored-by: León Orell Valerian Liehr <me@fmease.dev>

fleetingbytes committed:
Remove redundant empty line

one is enough

fleetingbytes committed:
Preserve potential mood for equal or NUL sign

Original `var_os` description said that it _may_ return an error if the value contains `=` or NUL. Let's make no promises on the `None` return value in these situation either, keep it in the [potential mood](https://en.wikipedia.org/wiki/Grammatical_mood#Potential).

cuviper commented on library/std/src/env.rs:
```diff
@@ -236,22 +236,14 @@ fn _var(key: &OsStr) -> Result<String, VarError> {
 
 /// Fetches the environment variable `key` from the current process, returning
-/// [`None`] if the variable isn't set or there's another error.
+/// [`None`] if the environment variable isn't set. It may return `None` also
```
Note, the first paragraph is best as a relatively short "title", as that's what will be listed on the module page. Can you break this up?

fleetingbytes committed:
Break up long first paragraph

Further referring to `var_os` as a "function" (like in `var`), rather than "method".

bors commented:
:pushpin: Commit 7d269633b1e254ad21f23994bbf04a6a8296abf7 has been approved by `cuviper`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=cuviper 7d269633b1e254ad21f23994bbf04a6a8296abf7 -->
<!-- homu: {"type":"Approved","sha":"7d269633b1e254ad21f23994bbf04a6a8296abf7","approver":"cuviper","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
